src/queue.rs

```diff
@@ -334,7 +333,7 @@ impl<'b, M: GuestAddressSpace> Iterator for AvailIter<'b, M> {
         *self.next_avail += Wrapping(1);
 
         Some(DescriptorChain::new(
-            self.mem.clone(),
+            self.mem,
```

Comment:
Removing this wouldn't work with the hotplug-friendly GuestAddressSpace implementations in vm-memory.

Reply:
Specifically for the example in this commit, it works based on the memory snapshot being created and held somewhere outside for the lifetime of the `Queue` object (that only gets the resulting `&GuestMemory` reference), and it looks like the overall idea remains the same even if we turn `M` into a smart pointer. I'll think a bit more about all these things (and a couple of others encountered while experimenting with queues) and update the proposal.

Reply:
The lifetime of the Queue is one of the following: 1) from one configuration change (including memory map changes) to the next or 2) from one ring processing until when the ring is fully processed.

In either case, the request (and thus the DescriptorChain) can outlive the Queue, and therefore the clone is necessary.

Reply:
(You can use `Arc<Queue>` instead; in that case you can use `&GuestMemory` as the argument. But it seems wrong to me when the actual thing that has an unknown lifetime is the memory map).

Reply:
Yeah, so the nuance the a `DescriptorChain` can be "stored" and outlive the `Queue` was not part of my original thinking. Will keep that in mind.

Reply:
To give this some context, having `DescriptorChain`s not tied to their `Queue`'s lifetime is really useful for things like servicing requests from a thread pool.

Reply:
Exactly, we ran into the same requirement when trying to adding multi-threading for virtio-fs backends.
It would be great to support Queue outlives the virtio event handler.